src/Resolvers/Collections.ts

```diff
@@ -92,31 +92,42 @@ export class CollectionsResolver {
 
   @FieldResolver(type => [Collection])
   async relatedCollections(
+    @Arg("size", () => Int, { nullable: true }) size: number,
     @Root() collection: Collection
   ): Promise<Collection[]> {
+    const query: { where: {}; take?: number } = { where: {} }
     const isArtistBased =
       collection.query.artist_ids && collection.query.artist_ids.length > 0
 
+    if (size) {
+      query.take = size + 1
+    }
+
     if (isArtistBased) {
-      const artistResults = await this.repository.find({
-        where: { "query.artist_ids": { $in: collection.query.artist_ids } },
-      })
+      query.where["query.artist_ids"] = {
+        $in: collection.query.artist_ids,
+      }
+
+      const artistResults = await this.repository.find(query)
 
       const relatedArtistCollections = reject(artistResults, {
         id: collection.id,
       })
 
+      // We only want to return relatedArtistCollections when there are more
+      // than 4. Otherwise we return category based related collections.
       if (relatedArtistCollections.length > 4) {
         return relatedArtistCollections
       }
     }
 
-    const relatedCategoryResults = await this.repository.find({
-      where: {
-        category: { $in: [collection.category] },
-        show_on_editorial: true,
-      },
-    })
+    query.where = {
+      category: { $in: [collection.category] },
+      show_on_editorial: true,
+    }
+
+    const relatedCategoryResults = await this.repository.find(query)
+
     const relatedCategoryCollections = reject(relatedCategoryResults, {
```

Comment:
I'm having a hard time describing this, so reach out via Slack if I'm not able to communicate well enough. 

I think we might need to add something like `.take(size)` to the end of this statement, in case the original collection was *not* included in the query results.

Example: I am working with collection 'photography'. I request 3 related collections, by sending `size: 3` as an argument. [We query for the first 4](https://github.com/artsy/kaws/pull/196/files#diff-500113cce4e1c0ce1050b25c61eb35a4R103) elements. We then [`reject` the 'photography' collection](https://github.com/artsy/kaws/pull/196/files#diff-500113cce4e1c0ce1050b25c61eb35a4R113) so that it's not included in its own "related" collections. But it turns out that none of the first 4 related collections were 'photography', so I still return 4 results, despite the client asking for only 3.

Reply:
Oh tricky! 

I think this is another point in favor of Sarah's suggestion to simply use the Mongo query to filter out the originating collection, something like:

```js
query.where = {
  id: { $ne: collection.id},
  category: collection.category,
  show_on_editorial: true,
 }
```

(Please check my Mongo syntax 😅)

That avoids the +1 problem and sidesteps any need for post-filtering with `reject`.